### PR TITLE
pkg: construct base libraries

### PR DESCRIPTION
**Pull request by IANTHEREAL**
We should build our base library and open it, like the feature that splits a table into chunks, etc.
It's good to reuse these library to develop some small tools.
I'm going to start with this PR and break up some of the libraries of TiDB-Binlog firstly. 

@july2993 @WangXiangUSTC PTAL

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
I fix vendor in pr #25

**Comment by IANTHEREAL** [comment]
ping @WangXiangUSTC @july2993

File: pkg/utils/tso.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"time"
+)
+
+// TSOToTime translates tso to timestamp
```
Comment (WangXiangUSTC) [inline pkg/utils/tso.go]: s/timestamp/time/

**Comment by WangXiangUSTC** [comment]
LGTM

File: pkg/utils/tso.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+package utils
+
+import (
+	"time"
+)
+
+// TSOToTime translates tso to time
+func TSOToTime(ts int64) time.Time {
+	t := time.Unix(ts>>18/1000, 0)
```
Comment (july2993) [inline pkg/utils/tso.go]: what about to be more precise get the Millisecond part to

Comment (IANTHEREAL) [reply pkg/utils/tso.go]: I think @WangXiangUSTC want a fuzzy time here, I would add comments or change it.  But anther problem, do we need a time zone in `generate savepoint`?

Comment (WangXiangUSTC) [reply pkg/utils/tso.go]: still need time zone, we can retain the code now, I will refine it next pr.

Comment (IANTHEREAL) [reply pkg/utils/tso.go]: I refine it later

File: generate_binlog_position/meta.go [hunk 3]
```diff
@@ -94,7 +94,7 @@ func (lm *localMeta) Save(ts int64, timeZone string) error {
 	}
 
 	if timeZone != "" {
-		t := pkg.TsToTime(ts)
+		t := pkg.TsToRoughTime(ts)
```
Comment (IANTHEREAL) [inline generate_binlog_position/meta.go:97]: I would remove `generate_binlog_position` in next pr, whatever

**Comment by IANTHEREAL** [comment]
@july2993 PTAL

**Review by july2993 — APPROVED** [review]
LGTM

**Pull request merged**